crates/router/src/compatibility/stripe/errors.rs

```diff
@@ -198,6 +198,8 @@ pub enum StripeErrorCode {
     FileNotAvailable,
     #[error(error_type = StripeErrorType::HyperswitchError, code = "", message = "There was an issue with processing webhooks")]
     WebhookProcessingError,
+    #[error(error_type = StripeErrorType::InvalidRequestError, code = "payment_method_unactivated", message = "The operation cannot be performed as the payment method used has not been activated. Activate the payment method in the Dashboard, then try again.")]
+    PaymentMethodUnactivated,
```

Comment:
An inactive payment method is just one of the factors involved in an eligibility analysis failure. There can be multiple other factors like
- amount exceeding maximum amount
- amount less than minimum amount
- payment method not supported by the connector
- capture method not supported by the connector
- 3DS not supported by the connector
etc.

Pinpointing the exact issue in eligibility analysis is a hard problem since it can be one factor or a combination of n factors for every connector of the merchant. Solving this problem is currently in the future scope, so we cannot pinpoint a particular factor for the failure. For now, it will be a general `EligibilityAnalysisFailure` only

Reply:
But this should not be a 5xx error, ( which it is currently ). The exact error should be propogated to the user / caller. In the current case the only way to check if something happened is to check logs and mca configuration

Reply:
The error type variant is changed to `IncorrectpaymentMethodConfigured`

Reply:
Can we make this change in Stripe Compatibility as well?

Reply:
Stripe does not have this error type, so we map it to the available error in stripe